### DISCUSSION TITLE
How to load mods with dependencies properly?

### DISCUSSION THREAD

**Discussion by MAZ12211** [FunkinCrew/Funkin#3948] [Q&A] [answered]
Here's the mod that needs dependency to work
```json
{
  "title": "random pack",
  "description": "r.",
  "api_version": "0.5.3",
  "mod_version": "1.0.0",
  "license": "Apache-2.0",
  "dependencies": {
    "SGRC - SilvaGunner tool": ">=1.0.0"
  }
}
```
I already have that dependency
Here's the Polymod metadata for it
```json
{
  "title": "SGRC - SilvaGunner tool",
  "description": "SGRC.",
  "api_version": "0.5.3",
  "mod_version": "1.0.0",
  "license": "Apache-2.0"
}
```

I have both in the mods folder yet the `random pack` mod doesn't even work in the game (V0.5.3)
![image](https://github.com/user-attachments/assets/96c668ae-1777-4adf-b804-c817480b7df4)

**Comment by NotHyper-474** [comment, marked as answer]
The dependency name needs to be the same as the name of the folder, not the title. So it should be:
```diff
{
  "title": "random pack",
  "description": "r.",
  "api_version": "0.5.3",
  "mod_version": "1.0.0",
  "license": "Apache-2.0",
  "dependencies": {
-     "SGRC - SilvaGunner tool": ">=1.0.0"
+     "SGRipperCredits": ">=1.0.0"
  }
}
```

**Reply by lunacynne** [reply]
> The dependency name needs to be the same as the name of the folder, not the title. So it should be:
> ```diff
> {
>   "title": "random pack",
>   "description": "r.",
>   "api_version": "0.5.3",
>   "mod_version": "1.0.0",
>   "license": "Apache-2.0",
>   "dependencies": {
> -     "SGRC - SilvaGunner tool": ">=1.0.0"
> +     "SGRipperCredits": ">=1.0.0"
>   }
> }
> ```

imo it should be the other way around...
i noticed gamebanana adds a hash to mod filenames which is kinda inconvenient for this since players have to know to change the folder name to whatever the dependency requires instead of it just... working according to the names in the json
